Testcontainer is not stopped, if class only contains @Disabled methods

# Description

When you have a test-class with only disabled test-methods, Testflight "forgets" to stop the corresponding container.

# Setup

If I have a class with only methods and that are annotated with `@Disabled`, the subsequent tests that use testflight fail, due to the container not being killed (see "The issue" for a log and my understanding of the problem).

## Example code

The following code fails:

```java
@MyQuarkusTest
class MyFirstListenerIT {

    @Test
    @Disabled // FIXME: remove, once implemented
    void testMyFirstListener() { }
}
```

On the other hand, I could fix this issue by putting the `@Disabled` on the class:

```java
@MyQuarkusTest
@Disabled // FIXME: remove, once implemented
class MyFirstListenerIT {

    @Test
    void testMyFirstListener() { }
}
```

A proper MWE might follow, if needed.

# The issue

My idea for the problem is, that if only a test-method is annotated with `@Disabled` and not the class, then Testflight does not notice, that this test was skipped. At the same time Testflight won't get a lifecycle callback that the test is finished and thus "forgets" to stop the container.
The workaround works, because when the test-class is disabled, then Testflight won't even start a container.

Also the bug *does not* occur, if there are other non-disabled test-methods.

Here's a log from our pipeline, where you can see two subsequent IT failing, due to not being able to assign the port. On my machine I could verify, that the container was still running.

```bash
[INFO] Running com.example.testflight.MyFirstListenerIT
2021-06-29 09:12:31,532 INFO  [🐳 [postgres:testflight-V1_0100]] (main) Creating container for image: postgres:testflight-V1_0100
2021-06-29 09:12:31,907 INFO  [🐳 [postgres:testflight-V1_0100]] (main) Starting container with ID: 342b63ee278cecc43f675b10713ffb6e2191825eb20c5e12f805f98fc1fc9504
2021-06-29 09:12:32,720 INFO  [🐳 [postgres:testflight-V1_0100]] (main) Container postgres:testflight-V1_0100 is starting: 342b63ee278cecc43f675b10713ffb6e2191825eb20c5e12f805f98fc1fc9504
2021-06-29 09:12:34,006 INFO  [🐳 [postgres:testflight-V1_0100]] (main) Container postgres:testflight-V1_0100 started in PT2.474243S

... Quarkus startup logs omitted...

[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 13.346 s - in com.example.testflight.MyFirstListenerIT
[INFO] Running com.example.testflight.MySecondListenerIT
2021-06-29 09:12:44,885 INFO  [🐳 [postgres:testflight-V1_0100]] (main) Creating container for image: postgres:testflight-V1_0100
2021-06-29 09:12:45,440 ERROR [🐳 [postgres:testflight-V1_0100]] (main) Could not start container: com.github.dockerjava.api.exception.InternalServerErrorException: Status 500: {"message":"driver failed programming external connectivity on endpoint blissful_franklin (77f2b7496ca0b7f8231d30f6c1ca0057c1bda65a3b7c461c78ce63e7bdf50d12): Bind for 0.0.0.0:32770 failed: port is already allocated"}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:247)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:102)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:31)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.StartContainerCmdImpl.exec(StartContainerCmdImpl.java:43)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:419)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
	at space.testflight.FlywayExtension.initialize(FlywayExtension.java:212)
	at space.testflight.FlywayExtension.beforeAll(FlywayExtension.java:92)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$8(ClassBasedTestDescriptor.java:368)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:368)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:192)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:136)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)

2021-06-29 09:12:45,453 ERROR [🐳 [postgres:testflight-V1_0100]] (main) There are no stdout/stderr logs available for the failed container
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.576 s <<< FAILURE! - in com.example.testflight.MySecondListenerIT
[ERROR] com.example.testflight.MySecondListenerIT  Time elapsed: 0.576 s  <<< ERROR!
org.testcontainers.containers.ContainerLaunchException: Container startup failed
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
Caused by: com.github.dockerjava.api.exception.InternalServerErrorException: 
Status 500: {"message":"driver failed programming external connectivity on endpoint blissful_franklin (77f2b7496ca0b7f8231d30f6c1ca0057c1bda65a3b7c461c78ce63e7bdf50d12): Bind for 0.0.0.0:32770 failed: port is already allocated"}


[INFO] Running com.example.testflight.MyThirdListenerIT
2021-06-29 09:12:45,479 INFO  [🐳 [postgres:testflight-V1_0100]] (main) Creating container for image: postgres:testflight-V1_0100
2021-06-29 09:12:45,667 INFO  [🐳 [postgres:testflight-V1_0100]] (main) Starting container with ID: 11125ca83ff2f6014f76553f373cf4f3e5e8ce9ebbd06034e5e78cc89624b7b7
2021-06-29 09:12:45,977 ERROR [🐳 [postgres:testflight-V1_0100]] (main) Could not start container: com.github.dockerjava.api.exception.InternalServerErrorException: Status 500: {"message":"driver failed programming external connectivity on endpoint crazy_kapitsa (7ee57d2f8e31c03091d4dd711a2cf159f0bde79b42a5fb11f1113d40cc81b312): Bind for 0.0.0.0:32770 failed: port is already allocated"}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:247)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.post(DefaultInvocationBuilder.java:102)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:31)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.StartContainerCmdExec.execute(StartContainerCmdExec.java:13)
	at org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at org.testcontainers.shaded.com.github.dockerjava.core.command.StartContainerCmdImpl.exec(StartContainerCmdImpl.java:43)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:419)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:325)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:323)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:311)
	at space.testflight.FlywayExtension.initialize(FlywayExtension.java:212)
	at space.testflight.FlywayExtension.beforeAll(FlywayExtension.java:92)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$8(ClassBasedTestDescriptor.java:368)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:368)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:192)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:136)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)

2021-06-29 09:12:45,979 ERROR [🐳 [postgres:testflight-V1_0100]] (main) There are no stdout/stderr logs available for the failed container
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.506 s <<< FAILURE! - in com.example.testflight.MyThirdListenerIT
[ERROR] com.example.testflight.MyThirdListenerIT  Time elapsed: 0.506 s  <<< ERROR!
org.testcontainers.containers.ContainerLaunchException: Container startup failed
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
Caused by: com.github.dockerjava.api.exception.InternalServerErrorException: 
Status 500: {"message":"driver failed programming external connectivity on endpoint crazy_kapitsa (7ee57d2f8e31c03091d4dd711a2cf159f0bde79b42a5fb11f1113d40cc81b312): Bind for 0.0.0.0:32770 failed: port is already allocated"}
```

for reference our `MyQuarkusTest` annotation looks like this:
```java
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Stereotype
@ExtendWith({FlywayExtension.class, QuarkusTestExtension.class})
@Flyway(dockerImage = "postgres:12",
        databaseInstance = Flyway.DatabaseInstanceScope.PER_TEST_METHOD,
        configuration = {
                @ConfigProperty(key = "space.testflight.jdbc.url.property", value = "quarkus.datasource.jdbc.url"),
                @ConfigProperty(key = "space.testflight.jdbc.username.property", value = "quarkus.datasource.username"),
                @ConfigProperty(key = "space.testflight.jdbc.password.property", value = "quarkus.datasource.password"),
                @ConfigProperty(key = "flyway.locations", value = "filesystem:src/main/resources/db/migration")
        })
public @interface MyQuarkusTest {
}
```

--- a/pom.xml
+++ b/pom.xml
@@ -356,7 +356,7 @@
       <plugin>
         <groupId>org.jacoco</groupId>
         <artifactId>jacoco-maven-plugin</artifactId>
-        <version>0.8.5</version>
+        <version>0.8.7</version>
         <executions>
           <execution>
             <id>default-prepare-agent</id>

--- a/pom.xml
+++ b/pom.xml
@@ -163,6 +163,11 @@
     </dependency>
     <dependency>
       <groupId>org.junit.platform</groupId>
+      <artifactId>junit-platform-testkit</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.platform</groupId>
       <artifactId>junit-platform-launcher</artifactId>
     </dependency>
     <dependency>

--- a/src/main/java/space/testflight/FlywayExtension.java
+++ b/src/main/java/space/testflight/FlywayExtension.java
@@ -127,6 +127,8 @@ public class FlywayExtension implements BeforeAllCallback, BeforeEachCallback, B
     if (getDatabaseInstance(context) == Flyway.DatabaseInstanceScope.PER_TEST_CLASS) {
       teardownDb(context, false);
     }
+
+    getGlobalStore(context).get(STORE_CONTAINER, AutoCloseable.class).close();
   }
 
   @Override

--- a/src/test/java/space/testflight/lifecycle/AbstractJupiterTestEngineTests.java
+++ b/src/test/java/space/testflight/lifecycle/AbstractJupiterTestEngineTests.java
@@ -1,0 +1,81 @@
+/*
+ * Copyright 2021 Marvin Kienitz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package space.testflight.lifecycle;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectMethod;
+import static org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder.request;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.engine.JupiterTestEngine;
+import org.junit.platform.engine.DiscoverySelector;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.UniqueId;
+import org.junit.platform.engine.discovery.ClassSelector;
+import org.junit.platform.engine.discovery.DiscoverySelectors;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.testkit.engine.EngineExecutionResults;
+import org.junit.platform.testkit.engine.EngineTestKit;
+
+public class AbstractJupiterTestEngineTests {
+  private final JupiterTestEngine engine = new JupiterTestEngine();
+
+  protected EngineExecutionResults executeTestsForClasses(Class<?>... testClasses) {
+    final ClassSelector[] classSelectors = Arrays.stream(testClasses)
+      .map(DiscoverySelectors::selectClass)
+      .collect(Collectors.toList())
+      .toArray(new ClassSelector[]{});
+
+    return executeTests(classSelectors);
+  }
+
+  protected EngineExecutionResults executeTestsForClass(Class<?> testClass) {
+    return executeTests(selectClass(testClass));
+  }
+
+  protected EngineExecutionResults executeTests(DiscoverySelector... selectors) {
+    return executeTests(request().selectors(selectors).build());
+  }
+
+  protected EngineExecutionResults executeTests(LauncherDiscoveryRequest request) {
+    return EngineTestKit.execute(this.engine, request);
+  }
+
+  protected TestDescriptor discoverTests(DiscoverySelector... selectors) {
+    return discoverTests(request().selectors(selectors).build());
+  }
+
+  protected TestDescriptor discoverTests(LauncherDiscoveryRequest request) {
+    return engine.discover(request, UniqueId.forEngine(engine.getId()));
+  }
+
+  protected UniqueId discoverUniqueId(Class<?> clazz, String methodName) {
+    TestDescriptor engineDescriptor = discoverTests(selectMethod(clazz, methodName));
+    Set<? extends TestDescriptor> descendants = engineDescriptor.getDescendants();
+    // @formatter:off
+    TestDescriptor testDescriptor = descendants.stream()
+      .skip(descendants.size() - 1)
+      .findFirst()
+      .orElseGet(() -> fail("no descendants"));
+    // @formatter:on
+    return testDescriptor.getUniqueId();
+  }
+}

--- a/src/test/java/space/testflight/lifecycle/DisabledMethodsTest.java
+++ b/src/test/java/space/testflight/lifecycle/DisabledMethodsTest.java
@@ -1,0 +1,71 @@
+/*
+ * Copyright 2021 Marvin Kienitz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package space.testflight.lifecycle;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.testkit.engine.EngineExecutionResults;
+
+import space.testflight.ConfigProperty;
+import space.testflight.Flyway;
+
+public class DisabledMethodsTest extends AbstractJupiterTestEngineTests {
+
+  @Test
+  public void executeTestsWithDisabledTestClass() {
+    assertThatNoException().isThrownBy(() -> {
+      EngineExecutionResults results = executeTestsForClasses(
+        TestClassWithOnlyDisabledTestMethodsTestCase.class,
+        TestClassWithNonDisabledTestMethodsTestCase.class
+      );
+
+      assertThat(results.testEvents().started().count()).isEqualTo(1);
+      assertThat(results.testEvents().skipped().count()).isEqualTo(1);
+      assertThat(results.testEvents().succeeded().count()).isEqualTo(1);
+      assertThat(results.testEvents().failed().count()).isZero();
+    });
+  }
+
+  @Flyway(
+    database = Flyway.DatabaseType.POSTGRESQL,
+    databaseInstance = Flyway.DatabaseInstanceScope.PER_TEST_METHOD,
+    configuration = {
+    @ConfigProperty(key = "space.testflight.jdbc.url.property", value = "javax.persistence.jdbc.url"),
+    @ConfigProperty(key = "space.testflight.jdbc.username.property", value = "javax.persistence.jdbc.user"),
+    @ConfigProperty(key = "space.testflight.jdbc.password.property", value = "javax.persistence.jdbc.password")
+  })
+  public static class TestClassWithOnlyDisabledTestMethodsTestCase {
+    @Test
+    @Disabled
+    public void testMethodDisabled() { }
+  }
+
+  @Flyway(
+    database = Flyway.DatabaseType.POSTGRESQL,
+    databaseInstance = Flyway.DatabaseInstanceScope.PER_TEST_METHOD,
+    configuration = {
+    @ConfigProperty(key = "space.testflight.jdbc.url.property", value = "javax.persistence.jdbc.url"),
+    @ConfigProperty(key = "space.testflight.jdbc.username.property", value = "javax.persistence.jdbc.user"),
+    @ConfigProperty(key = "space.testflight.jdbc.password.property", value = "javax.persistence.jdbc.password")
+  })
+  public static class TestClassWithNonDisabledTestMethodsTestCase {
+    @Test
+    public void testMethodNonDisabled() { }
+  }
+}

